### ISSUE TITLE
[BUG]: The progress bar stops at a certain percentage and warm_start doesn't seem to be working

### ISSUE THREAD

**Issue by Angelld23** [MilesCranmer/PySR#953]
### What happened?

Hello

Every time I start a new run, the model start having progress for a couple of minutes, but after 30 minutes or around 11% of progress it stops improving. I tried leaving it there for a couple hours to see if it just that the search time increased exponentially, but it doesn't look like. I would like to know if this might be a bug or maybe is a problem of my laptop. Also, When I try to run again starting from the last checkpoint (warm_start=True) it doesn't look like it works because it starts from a new checkpoint with different equations with higher loss.

System: Hp Laptop

Edition: Windows 11 Enterprise
Version: 24H2
Processor: 13th Gen Intel(R) Core(TM) i7-1360P   2.20 GHz
RAM: 32.0 GB (31.6 GB usable)
System Type: 64-bit operating system, x64-based processor

Python: 3.11.9
PySR version: 1.5.8

Code running on Jupyter Kernel (notebook) in VS-Code

### Version

1.5.8

### Operating System

Windows

### Package Manager

pip

### Interface

Jupyter Notebook

### Relevant log output

```shell

```

### Extra Info

loss_function = """
function f(tree, dataset::Dataset{T,L}, options) where {T,L}
    ypred, completed = eval_tree_array(tree, dataset.X, options)
    if !completed
        return L(Inf)
    end
    y = dataset.y
    return sqrt(sum(i -> (ypred[i] - y[i])^2, eachindex(y)))
end
"""

model = PySRRegressor(
    niterations=200,  # < Increase me for better results
    ncycles_per_iteration = 380,
    populations=250,
    population_size=250,
    binary_operators=["*", "/", "-", "+"],
    unary_operators=[
        "exp",
        "sqrt",
        "square",
        "log10",
        "cube",
    ],
    # ^ Define operator for SymPy as well
    warm_start=True,
    maxsize=40,
    loss_function=loss_function,
    adaptive_parsimony_scaling =1000,
    fraction_replaced_hof =0.1,
    select_k_features = 5,
    parsimony = 0.0001,
    weight_optimize = 0.001,
    nested_constraints = {
        "exp": {"exp": 0},
        "sqrt": {"sqrt": 0},
        "square": {"square": 0},
        "log10": {"log10": 0},
        "cube": {"cube": 0},
    },
    output_directory=r'C:\Users\xxxxxx',
)
model.fit(X_train, y_train)
